### PR TITLE
Improve virus summary

### LINKED ISSUE #457: Improve group-summary-viruses.py
It would be good if:

* The data collection / processing and the output functionality in this script were separated. The former could even have tests.
* There was a command line `--html` argument to produce HTML output (then `group-summary-viruses-html.py` can go away).
* There was a `--filenameRegex` argument that could be used to extract a part of each incoming filename for the purposes of grouping files. If not given, individual files are just considered to be individual samples (the way it works now). With this argument, we could pass a regex like `_(DA\d+)_` to pull human sample identifiers out of Eske Willerslev files.
* I find the file names `group-summary-viruses.py` and `group-summary-proteins.py` very confusing.
* Optionally, it would be nice to collect all the reads (per virus per sample), remove duplicates, and write them to a file.  If we don't have a way to do this it has to be done manually by anyone wanting to do further work on the reads that match a virus for a sample.  This doesn't have to be done in this ticket, but at least should be kept in mind.

### PR DESCRIPTION
**Pull request by terrycojones**
This addresses all the issues in #457, apart from the grouping of FASTA files which I'd rather do in a later/separate branch.

Fixes #457.

### CODE REVIEW

File: bin/proteins-to-viruses.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import argparse
+import sys
+
+from dark.proteins import ProteinGrouper
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="Group proteins by the virus they're from")
```
Comment (bamueh) [inline bin/proteins-to-viruses.py]: "Group proteins by the virus they're from."

Comment (bamueh) [inline bin/proteins-to-viruses.py]: Also, why are there double quotes?

File: bin/proteins-to-viruses.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    parser.add_argument(
+        'filenames', nargs='*', help='Sample file names to read input from')
```
Comment (bamueh) [inline bin/proteins-to-viruses.py]: 'Sample file names to read input from.'

File: dark/proteins.py [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+                'medianScore': float(medianScore),
+                'proteinLength': int(proteinLength),
+                'proteinTitle': proteinTitle,
+                'proteinURL': NCBISequenceLinkURL(proteinTitle),
+                'readCount': int(readCount),
+            })
+
+    def toStr(self):
+        """
+        """
```
Comment (bamueh) [inline dark/proteins.py]: Add something.

File: dark/proteins.py [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+                        '%(bestScore).2f\t%(readCount)4d\t%(hspCount)4d\t'
+                        '%(index)3d\t%(proteinTitle)s'
+                        % proteinMatch)
+            append('')
+
+        return '\n'.join(result)
+
+    def toHTML(self):
+        """
+        """
```
Comment (bamueh) [inline dark/proteins.py]: Same.

**Comment by bamueh** [comment]
Looks good.

File: bin/proteins-to-viruses.py (hunk 1 above)
Comment (terrycojones) [inline bin/proteins-to-viruses.py]: Fixed (first one). I usually use "" when the string contains a single quote.

File: dark/proteins.py (hunk 3 above)
Comment (terrycojones) [inline dark/proteins.py]: Done.

**Pull request merged**